spec/controllers/api/items_controller_spec.rb

```diff
@@ -112,6 +112,29 @@
             items_param = serialize_to_hash(items_param)
             expect(saved_items).to match_array(items_param)
           end
+          it 'should store revisions matching changes' do
+            @controller = Api::AuthController.new
+            post :sign_in, params: test_user_credentials
+
+            @controller = Api::ItemsController.new
+            request.headers['Authorization'] = "bearer #{JSON.parse(response.body)['token']}"
+
+            # Serializing the items into an array of hashes
+            items_param = test_items.limit(1).to_a.map(&:serializable_hash)
+
+            items_param[0]['content'] = 'This is the new content.'
+            post :sync, params: { sync_token: '', cursor_token: '', limit: 5, api: '20190520', items: items_param }
+
+            items_param[0]['content'] = 'This is yet another new content.'
+            post :sync, params: { sync_token: '', cursor_token: '', limit: 5, api: '20190520', items: items_param }
+
+            item = Item.where(uuid: items_param[0]['uuid']).first
+
+            revisions = item.revisions
+            expect(revisions.count).to eq(3)
```

Comment:
Wait why do we expect 3 and not 2? I'm only seeing 2 sync requests made 🤔

Reply:
It should be 1 revision for when the item is created, and 2 revisions for the sync requests above.

Reply:
1st revision is made upon the creation of an item, 2 following ones are the updates via sync